flask_hal/link.py

```diff
@@ -0,0 +1,248 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+"""
+flask_hal.link
+==============
+
+Implements the ``HAL`` Link specification.
+"""
+
+# Standard Libs
+import json
+
+
+VALID_LINK_ATTRS = [
+    'name',
+    'title',
+    'type',
+    'deprecation',
+    'profile',
+    'templated',
+    'hreflang'
+]
+
+
+class Links(object):
+    """Build a collection of ``HAL`` link objects.
+
+    Example:
+        >>> from flask_hal.link import Link, Links
+        >>> l = Links(
+        ...     Link('foo', 'http://foo.com'),
```

Comment:
A view with KIm and SQL Alchemy may look like this:

``` python
@app.route('/foo/:id')1
def foo_view(id):
    foo = Foo.query.get(id)
    data = FooSerializer().serialize(foo)

    document = Document(links=Links(foo.hal_link()), data=data)

    return document, 200
```

I'm still unsure on the `Document` class API.